athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java

```diff
@@ -196,20 +200,54 @@ else if (vector instanceof StructVector) {
      *
      * @param vector The FieldVector into which we should write the provided value.
      * @param pos The row number that the value should be written to.
-     * @param value The value to write.
+     * @param rawValue The value to write.
      * @note This method incurs more Object overhead (heap churn) than using Arrow's native interface. Users of this Utility
      * should weigh their performance needs vs. the readability / ease of use.
      */
-    public static void setValue(FieldVector vector, int pos, Object value)
+    public static void setValue(FieldVector vector, int pos, Object rawValue)
     {
         try {
-            if (value == null) {
+            if (rawValue == null) {
                 setNullValue(vector, pos);
                 return;
             }
+            /**
+             * We will convert any types that are not supported by setValue to types that are supported
+             * ex) (not supported) org.joda.time.LocalDateTime which is returned on read from vectors
+             * will be converted to (supported) java.time.ZonedDateTime
+             * @param o1 object with unsupported type
+             * @return another object with supported/converted type
+             */
+
+            Object value = rawValue;
+            if (rawValue instanceof org.joda.time.LocalDateTime) {
+                DateTimeZone dtz = ((org.joda.time.LocalDateTime) rawValue).getChronology().getZone();
+                long dateTimeWithTimeZone = ((org.joda.time.LocalDateTime) rawValue).toDateTime(dtz).getMillis();
+                value = DateTimeFormatterUtil.constructZonedDateTime(dateTimeWithTimeZone);
+            }
```

Comment:
Why do this for every single value? This check is irrelevant for 9 out of 10 types yet we'd take the branching perf hit for all of them. Can this be moved into the relevant switch statement path?